Server/mods/deathmatch/logic/CAccount.cpp

```diff
@@ -46,8 +46,9 @@ void CAccount::SetName(const std::string& strName)
 {
     if (m_strName != strName)
     {
-        m_pManager->ChangingName(this, m_strName, strName);
-        m_strName = strName;
+        std::string strNewName = strName.substr(0, MAX_USERNAME_LENGTH);
```

Comment:
We could avoid a copy here, if we change `CAccount::SetName` to return a boolean and check if the length is in our predefined boundaries.

Reply:
std::string_view could be used instead.
```cpp
auto strTrimmedName = std::string_view{strName}.substr(0, MAX_USERNAME_LENGTH);
```

Reply:
Actually:
```cpp
if (strName.length() > MAX_USERNAME_LENGTH)
    strName.erase(MAX_USERNAME_LENGTH - 1);
```

Reply:
Let's not do that before 1.6. Breaks scripts that are currently (wrongly) depending on overlong usernames in their code.